### ISSUE TITLE
Deeply nested includes

### ISSUE THREAD

**Issue by andcl85** [flugger/laravel-responder#33]
I don't know if its a bug or if I am not doing it well, but I have a 3-level deep model relationship, i.e., **Survey** that has many **Items** which, in turn, have many **Results**.
I want to respond with nested information about the 3 levels, having 'items' and 'results' methods in 'SurveyTransformer' and 'ItemTransformer', respectively, but doing things like...

`...new SurveyTransformer) ->include(['items.results']) -> respond();`
or 
`...new SurveyTransformer) ->include(['items', 'results']) -> respond();`
(my tries so far) does not seem to work...

It returns the 'items' info correctly, but nothing about the 'results', only a key `'items.results' = null` at the end of the items...
Can you please point me in the right direction?
Thank you very much in advance.

**Comment by flugger** [comment]
The `'items.results' = null` is actually a bug and something I'm addressing for the next release. You should, however, see results nested under items as well. Could you paste the `results` method in your ItemTransformer?

**Comment by andcl85** [comment]
Of course, 
```
public function results(Item $item)
{
    return $this -> collection($item -> results, new ResultTransformer);
}
```
Nothing special with this method, I guess...
What is remarkable is that the key `'items.results' = null` only appears one time, at the end of the last item...
Thanks.

**Comment by flugger** [comment]
Hmm, I'm working on a rewrite of the package improving the including of relations, which should solve these issues. I'll try to recreate the issue described above, but could you try to change the `results` method to the following:

```
public function results(Item $item) {
    return responder()->transform($item->results, new ResultTransformer)->getResource();
}
```

**Comment by andcl85** [comment]
Of course.
I have modified the method with your suggestion and I get the same `'items.results' = null` key at the end...
(Assuming that in the controller I have:
`...new SurveyTransformer) ->include(['items.results']) -> respond();`
)

**Comment by flugger** [comment]
Hey, sorry for taking so long. After some investigating, I've located the bug. The issue is when using the wildcard relations on deeply nested transformers, like:

```php
protected $relations = ['*'];
```

For some reason, that doesn't work as intended for nested relations. However, it can be fixed by explicitly specifying the relationship, which in your case would be to add `$relations = ['results']` to `ItemTransformer`. 

I tested the same with the upcoming version 2 of the package, and everything worked as intended. So this should be solved by that in a couple weeks time - in the mean time, you will have to be explicit with the relations (which often can be a good idea anyway, as it can restrict access to some relations for security reasons).

Let me know if that solved all your issues!

**Comment by andcl85** [comment]
Thank you very much for your help, but that change does not fix the issue in my specific case...
I have done the changes to `ItemTransformer` so that `protected $relations = ['results']`, and it keeps returning the `'items.results' = null` at the end.
I have also tried with both

`return responder()->transform($item->results, new ResultTransformer)->getResource();`
and 
`return $this -> collection($item -> results, new ResultTransformer);`
in the `results` method, no luck...

However, noticed that when I change `protected $relations = ['items']` in `SurveyTransformer`, `results` does not appear, but neither `'items.results' = null` key.

Thanks again.

**Comment by flugger** [comment]
Oh, I thought the issue was that you didn't get the `'results'` data at all. Is this still the case? The `'items.results': null` at the end is a bug I tried to fix earlier this winter. However, while trying to fix it, I realised the entire code for handling relations was way too complex, and instead of fixing it, I ended up rewriting big parts of the package (for the better). These changes will be rolled out soon and will fix the `'items.results': null` bug. In the meantime, the bug shouldn't actually _break_ anything, only add an unnecessary JSON field to the response.
